File: decompress-run-length-encoded-list.js
var decompressRLElist = function(nums) {
  let arr = [];
  for (let index = 0; index < nums.length; index += 2) {
    let count = nums[index];
    while (count-- > 0) {
      arr.push(nums[index + 1]);
    }
  }
  return arr;
};

File: decompress-run-length-encoded-list.rs
impl Solution {
    pub fn decompress_rl_elist(nums: Vec<i32>) -> Vec<i32> {
        let mut v = vec![];
        for i in (0..nums.len()).step_by(2) {
            let freq = nums[i];
            for j in 0..freq {
                v.push(nums[i + 1]);
            }
        }
        v
    }
}
